api/v1beta1/kafkacluster_types.go

```diff
@@ -161,18 +176,24 @@ type DisruptionBudgetWithStrategy struct {
 	DisruptionBudget `json:",inline"`
 	// The strategy to be used, either minAvailable or maxUnavailable
 	// +kubebuilder:validation:Enum=minAvailable;maxUnavailable
-	Stategy string `json:"strategy,omitempty"`
+	Strategy string `json:"strategy,omitempty"`
 }
 
 // Broker defines the broker basic configuration
 type Broker struct {
+	// id maps to "node.id" configuration in KRaft mode, and it maps to "broker.id" configuration in ZooKeeper mode.
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:ExclusiveMaximum=true
 	Id                int32         `json:"id"`
 	BrokerConfigGroup string        `json:"brokerConfigGroup,omitempty"`
 	ReadOnlyConfig    string        `json:"readOnlyConfig,omitempty"`
 	BrokerConfig      *BrokerConfig `json:"brokerConfig,omitempty"`
+	// processRoles defines the role(s) for this particular broker node: broker, controller, or both.
+	// This must be set in KRaft mode.
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Roles []string `json:"processRoles,omitempty"`
```

Comment:
An alternative option is to make this field as a part of the `BrokerConfig` struct. This way users can use `brokerConfigGroup` to set the process roles for the corresponding Kafka nodes.

But honestly, I can't really tell when we should put things under `BrokerConfig`, and when to put under `Broker`

Reply:
I'd agree with going with BrokerConfig instead as I can imagine that it might be useful in some (probably very few) scenarios to set a group of brokers as either of these roles, for example when trying to do some sort of rack awareness scenario or going for some sort of geological distribution maybe? I'm not sure if it's really worth the effort though or if it would even help in those scenarios.

Reply:
Another potential use case of moving this `Roles` under `BrokerConfig` - let's give all the Kafka nodes with this specific role certain amount of resources  - e.g. "controller" might need less resources than "brokers", and the "combined" nodes might need the most resources. This way, users can just use the `brokerConfigGroup` to make their life a tidy bit easier.

Reply:
I think i will make this change in the `KafkaCluster` API, and will update the main PR accordingly

Reply:
I made the change in https://github.com/banzaicloud/koperator/pull/1024/commits/54af7053ae40fae45bc2a6806b350b5c90436a53, reviewers - please let me know which version you think is better